synapse/api/auth.py

```diff
@@ -629,7 +629,13 @@ def get_user_by_access_token(self, token, rights="access"):
         except AuthError:
             # TODO(daniel): Remove this fallback when all existing access tokens
             # have been re-issued as macaroons.
+            if self.hs.config.expire_access_token:
+                raise AuthError(
+                    401,
+                    "access token has expired"
+                )
```

Comment:
We don't actually know if the `AuthError` was due to the macaroon having expired at this point. 

What was the reason for raising here? To give an appropriate error message? Or stopping it from falling back to checking if the access token is in the database?